### DISCUSSION TITLE
Do I dare upgrade?

### DISCUSSION THREAD

**Discussion by jason-barba** [GrandOrgue/grandorgue#901] [Help]
I run an old version of GO 0.3.1.2334+155. It works flawlessly on my computer and with my maunals, pedel and FCB1010 , not to meantion my touch screen. It runs on Ubuntu 20.04 LTS.

Can I expect a seamless upgrade to the latest GO version? How can I prepare for a roll back if anything should go wrong?

I depend on my organ ;-)

//Jens

**Comment by midimusic** [comment]
Upgrading to something new which is potentially buggy is always risky and time consuming. There is no point in upgrading if you are satisfied with what you already have.

There is no way to roll back other than by uninstalling the new version and reinstalling the old version. So I recommend you upgrade only if you have still got your original install program.

csw900

**Comment by oleg68** [comment]
@jason-barba You may to get some problems with midi settings, because the device names are changed between these versions.

The safest way of trying the new version is
1. Backup the `GrandOrgueConfig` file
2. Download the [tar.gz version of GO](https://github.com/GrandOrgue/grandorgue/releases/download/3.4.4-1/grandorgue-3.4.4-1.linux.x86_64.tar.gz) and unpack to some empty directory
3. Run GrandOrgue from this directory without installation. Make necessary changes in the settings
4. If you don't like the new version, just delete the directory and restore the old config file. The old version will be still installed
5. If you like the new version, then download the `.deb` package and reinstall it with `apt-get`.

**Reply by jason-barba** [reply]
Lifting this thread. I think I now have to take the step.

Is GrandOrgueConfig the only file I have to make backup of before doing step 3 above? What about the GrandOrgue folders? Are they not affected by running a new version of GO? Are the files in those folders only read by the trial GO version?

**Reply by ahall41** [reply]
Personally, I would make sure I have a backup of everything (using [CloneZilla](https://clonezilla.org/) or similar). Especially if you "depend on your organ". However, it should be sufficient to backup your organ settings (mine are in GrandOrgue/Data), and also export the settings for your favourite organs too (which go into the Settings folder - depends what your Paths are set to).

The main problem I found when upgrading is that the Midi device names have changed. This can be quite tedious; I got round it with a script that gunzipped every .cmb file, used "sed" to change the device name; and gzipped it back. But to quote larspolo (#700): "The. .. cmb files are not supposed to be edited by users by hand. Ever! As in never!"

You could also use the -i command line option ({pathtogo}/GrandOrgue -i xxx) which will create a separate instance of GrandOrgue (using GrandOrgueConfig-xxx, Data-xxx and Cache-xxx). Then, import settings for your organ from the original folder.

**Reply by jason-barba** [reply]
Thanks for kind advice.

Regarding the MIDI device names: Is there a way to know which names to change, from and to? I can understand that this is a problem if you have use several sample sets, but I only use a single one.

My guess is that I in any case have to manually assign my MIDI devices to the first (and in my case the only) sample set in order get the new MIDI device names. True?

**Reply by ahall41** [reply]
If you only have the one sample set, its probably easier to manually assign the devices. In my case I had lots of stops assigned, and quite a few sample sets. I would download the tar.gz version, and use the -i option (from the command prompt). That way you can get the new version working without upsetting the previous one.

**Reply by jason-barba** [reply]
I have 2 manuals and a pedal plus a FCB1010 foot controller. The latter has about ten+ buttons and swell pedals hooked up to generals/stops/combination ctrl.

I have downloaded an untarred a new GO version. Is the -i option used when starting GO from the command line? And what does it do?

**Reply by ahall41** [reply]
Yes use it from the command line; -i xxx will append this (anything you like for xxx - 3.6 maybe) to the paths shown above. You will then need to teach GrandOrgue the new settings. If you have any combinations stored, export them in GrandOrgue 3.1 and then import into GrandOrgue 3.6

**Reply by jason-barba** [reply]
I am sorry, It is still unclear to me what the option -i does. Is xxx the path to some file or folder?

**Reply by ahall41** [reply]
No. -i creates another (separate) instance. You can use anything you choose in place of xxx. The result of which it will put your configuration in GrandOrgueConfig-xxx instead of GrandOrgueConfigf, your settings in GrandOrgue/Data-xxx instead of GrandOrgue/Data, and your cache in GrandOrgue/Cache-xxx instead of GrandOrgue/Cache. GrandOrgue will create the files and folders for you. For example, If you downloaded and extracted the tar.gz archive to folder Downloads/grandorgue-3.6.2, Running from the command line `Downloads/grandorgue-3.6.2/bin/GrandOrgue -i 362` will give you GrandOrgueConfig-362, etc.

**Reply by jason-barba** [reply]
So a suggested WOW to test upgrade of GO should be:
1. Backup!
2. Start the test version of GO from command line with option -i (this creates required files and folders)
3. Close GO
4. Copy the content of all running GO folders and files to test folders.
5. Start test GO again and update MIDI settings

Correct? Have I missed something.

If all goes well, I can upgrade the running GO version using the deb package.

**Comment by jason-barba** [comment]
Aha! Thanks for explaining. Funny that I cannot find the command line options anywhere.
But: Where are these new directories created? In my home directory or in the directory where the tested version of GO is stored?

**Reply by oleg68** [reply]
In your home directory.

**Reply by jason-barba** [reply]
:-)

**Reply by jason-barba** [reply]
Could you please advice me where to find documentation on the CLI command and its options/switches. I have searched but with no success.

**Reply by ahall41** [reply]
Use -h or --help as an option ... not sure where/how I discovered it. There's only 1 other option: -i (or --instance) "specify GrandOrgue instance name". 
You can also pass an organ file on the command line (so you associate GrandOrgue with the .organ suffix).

**Comment by jason-barba** [comment]
So, a quick recap (Ubuntu 20.04 LTS):

1. Backup GrandOrgue* data
2. apt install <latest version>.deb
3. Check/Reconfigure MIDI device

Ok? The apt install overwrights the old version completely? Old version being 0.3.something.

//Jens

**Comment by jason-barba** [comment]
Tried upgrade to latest .deb distirbution in my "sandbox". First, got this error message as final printout on the console:
.
.
.
Processing triggers for desktop-file-utils (0.24-1ubuntu3) ...
N: Download is performed unsandboxed as root as file '/home/jens/Downloads/grandorgue_3.6.7-1_amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
jens@ElviraFossa:~$

Then started GO, which gave the log output as shown in attached file. Is this expected?
The Program Settings menu opened in tab MIDI Devices. Is this due to the renaming of MIDI Devices? And is the cure just to redo the MIDI setup (right click on manuals and stops and Listen for Event)?

//Jens

**Reply by ahall41** [reply]
I'm assuming GO now reports itself as 3.6.7.1? You didn't attach the log file, but redoing the MIDI set up should resolve things ... ?

**Reply by jason-barba** [reply]
Yes, it reports correct verison.
Sorry for the missing log, inclyuded here.
![Screenshot 2022-05-31 095230](https://user-images.githubusercontent.com/56149267/171152611-3116006b-7d91-42d5-abda-44e54f36b6b6.png)
//Jens

**Reply by ahall41** [reply]
You can ignore this warning - when you save your settings it should go away.

**Comment by jason-barba** [comment]
I didn't have to worry. The upgrade of GO happened automagically when I upgraded Ubuntu from 20.04 LTS to 22.04 LTS.
See https://github.com/GrandOrgue/grandorgue/discussions/1183
